test/SerialIOTest.cpp

```diff
@@ -414,6 +414,13 @@ TEST_CASE("available_chunks_test_json", "[serial][json]")
 
 TEST_CASE("multiple_series_handles_test", "[serial]")
 {
+    /*
+     * clang also understands these pragmas.
+     */
+#if defined(__GNUC__MINOR__) || defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
With this one set, I think you can remove `-Wno-deprecated-declarations` from quite a few CI scripts in `.github/workflows/`

Reply:
-> #1246